src/components/RequiresAuth/RequiresAuth.jsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Navigate, useLocation } from 'react-router-dom';
+import { useAuth } from '../../context/auth/authContext';
+
+function RequiresAuth({ children }) {
+  const { authState } = useAuth();
+  const location = useLocation();
+
+  return authState.token === null ? (
+    <Navigate to="/login" state={{ prevPath: location.pathname }} />
```

Comment:
you need `replace` here, without it, if I go to login page, then I cannot navigate back (since you have replace in login at line 22)